scripts/barclamp_install.rb

```diff
@@ -66,47 +69,42 @@ def rm_tmpdir(tmpdir)
   end
 end
 
+barclamp_yml_files = Array.new
+component_paths = Array.new
 ARGV.each do |src|
   debug "src: #{src}"
-  case
-  when /tar\.gz|tgz$/ =~ src
-    # This might be a barclamp tarball.  Expand it into a temporary location.
-    src=File.expand_path(src)
-    system "tar xzf \"#{src}\" -C \"#{tmpdir}\""
-    # get the barclamp name, allow the tar ball to be named <barclampname>-barclamp.t*
-    target="#{tmpdir}/#{src.split("/")[-1].split(".")[0].chomp("-barclamp")}"
-    if get_crowbar_yml_path(target).nil?
-      puts "#{src} is not a barclamp tarball, ignoring."
-    else
-      candidates << target
-    end
-  when !get_crowbar_yml_path(src).nil?
-    # We were handed something that looks like a path to a barclamp
-    candidates << File.expand_path(src)
-  when !get_crowbar_yml_path(File.join(BARCLAMP_PATH, src)).nil?
-    candidates << File.join(BARCLAMP_PATH, src)
+  if Dir.exist?(src)
+    barclamp_yml_files += get_yml_paths(src)
+    component_paths.push src
+  elsif Dir.exist?(File.join(BARCLAMP_PATH, src))
+    barclamp_yml_files += get_yml_paths(File.join(BARCLAMP_PATH, src))
+    component_paths.push File.join(BARCLAMP_PATH, src)
```

Comment:
There's something I'm not getting: if I do `barclamp_install.rb core` it will look in /opt/dell/barclamps/core. But the yml files are still in /opt/dell/barclamps/{crowbar,dns,ntp,...}/. How can this work? I assume I'm missing something :-)